product-scenarios/scenarios-commons/src/main/java/org/wso2/identity/scenarios/commons/util/IdentityScenarioUtil.java

```diff
@@ -88,6 +112,55 @@ public static HttpResponse sendGetRequest(HttpClient client, String url, Map<Str
         return client.execute(getRequest);
     }
 
+    /**
+     * Send GET request for a given URL with the request query parameters.
+     *
+     * @param client HttpClient to be used for request sending.
+     * @param url Request URL.
+     * @param params Request query parameters.
+     * @param headers Request headers.
+     * @return HttpResponse containing the response.
+     * @throws IOException If error occurs while sending the request.
+     * @throws URISyntaxException If error occurs while constructing the request URL.
+     */
+    public static HttpResponse sendGetRequest(HttpClient client, String url, Map<String, String> params,
```

Comment:
Can you also refactor the other `sendGetRequest` method to internally call this method with null headers ?